src/util/async-route.ts

```diff
@@ -0,0 +1,12 @@
+import express from 'express'
+
+export function asyncRoute(handler: express.Handler) {
+  return (
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction,
+  ) => {
+    // returns to make testing more straightforward. the output is not actually needed.
+    return Promise.resolve(handler(req, res, next)).catch(next)
```

Comment:
I feel like i should know this, but why do you need the `Promise.resolve`?

Reply:
so I have to admit here I kind of copied this over from where it's been used in IHL, since Satish linked to that in his PR comment asking for better handling of unhandled promise rejections. 

I think I see why it's good though. MDN says

> Promise.resolve() returns a promise whose eventual state depends on another promise, thenable object, or other value.

So whether the handler actually returns a promise or not, this is a safe way of adding a `catch`.

I *think* this is equivalent
```
try {
  await handler(req, res, next)
} catch (error) {
  next(error)
}
```
and the "await" would just do nothing when the `handler` doesn't actually return a promise. Maybe the try/catch is easier to reason about and we should just use that in the future.

Since the indexer is hopefully going to the dustbin soon, I'm going to merge as-is, but I'd be interested to hear what others think about this asyncHandler pattern since we've seen it come up a few times now.

Reply:
the one thing I wouldn't advocate for is just doing `handler(req, res, next).catch(next)`, since it just seems more error prone for the cases where `asyncHandler` is called with something that doesn't return a promise. Not sure why we'd call it then, but since it's easy to make the function a little safer, I prefer it that way.

Reply:
though maybe the ts compiler should protect us from that anyway...? idk, kinda going down a rabbit hole here

Reply:
ohhhh, the more robust handling of non-promise handlers makes sense.  i wonder if it's a pattern we copied from vanilla js.